src/App.jsx

```diff
@@ -13,16 +15,14 @@ class App extends React.Component {
   /**
    * @returns {undefined}
    */
-  componentDidMount() {
-    window.addEventListener('app-toast', event => {
-      if (event.detail.type === 'error') {
-        event.detail.messages.forEach(message => toast.error(message));
-      } else if (event.detail.type === 'success') {
-        event.detail.messages.forEach(message => toast.success(message));
-      } else {
-        event.detail.messages.forEach(message => toast.info(message));
-      }
-    });
+  async componentDidMount() {
+    const user = await checkAuth();
+    if (user) store.dispatch(authSuccessAction({ user }));
```

Comment:
Directly `dispatch`ing actions within a component is an antipattern; that is what `connect` is for.

Reply:
The `App` component has the `Provider` as a child component so I can't connect to the child. That's why it was done this way. I will make `Provider` the parent of `App` in `index.jsx`  and remove it from this file. That way, I'll be able to connect to the store from here.